stripe/src/main/java/com/stripe/android/paymentsheet/model/ClientSecret.kt

```diff
@@ -0,0 +1,24 @@
+package com.stripe.android.paymentsheet.model
+
+import android.os.Parcelable
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.SetupIntent
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Represents the client secret for a [SetupIntent] or [PaymentIntent]
+ */
+internal sealed class ClientSecret(
+    open val value: String
+) : Parcelable {
+
+    @Parcelize
+    data class PaymentIntentClientSecret(
+        override val value: String
+    ) : ClientSecret(value)
+
+    @Parcelize
+    data class SetupIntentClientSecret(
+        override val value: String
+    ) : ClientSecret(value)
```

Comment:
I think you should be able to move these outside of `ClientSecret`, so that you don't need the superverbose class name

```
sealed class ClientSecret

data class PaymentIntentClientSecret(...) : ClientSecret()

data class SetupIntentClientSecret(...) : ClientSecret()
```

Reply:
But then I can't get the value from a ClientSecret directly, and need to cast it first.

Reply:
You can if it's

```
sealed class ClientSecret {
  abstract val value: String
}
```